mobile-app/app/screens/AppNavigator/screens/Loans/LoansScreenV2.tsx

```diff
@@ -0,0 +1,206 @@
+import { useAppDispatch } from "@hooks/useAppDispatch";
+import { LoanParamList } from "./LoansNavigator";
+import { LoanCardsV2 } from "./components/LoanCardsV2";
+import { EmptyVault } from "./components/EmptyVault";
+import { Vaults } from "./components/Vaults";
+
+enum TabKey {
+  BrowseLoans = "BROWSE_LOANS",
+  YourVaults = "YOUR_VAULTS",
+}
+
+type Props = StackScreenProps<LoanParamList, "LoansScreen">;
+
+export function LoansScreenV2({ navigation }: Props): JSX.Element {
+  const { address } = useWalletContext();
+
+  const isFocused = useIsFocused();
+  const blockCount = useSelector((state: RootState) => state.block.count);
+  const { vaults, hasFetchedVaultsData, hasFetchedLoansData } = useSelector(
+    (state: RootState) => state.loans
+  );
+
+  const vaultsList = useSelector((state: RootState) =>
+    vaultsSelector(state.loans)
+  );
+  const loans = useSelector((state: RootState) =>
+    loanTokensSelector(state.loans)
+  );
+
+  const [activeTab, setActiveTab] = useState<string>(TabKey.YourVaults);
+  const dispatch = useAppDispatch();
+  const client = useWhaleApiClient();
+  const onPress = (tabId: string): void => {
+    if (tabId === TabKey.YourVaults) {
+      setShowSearchInput(false);
+    } else if (searchString !== "") {
+      setShowSearchInput(true);
+    } else {
+      // no-op: maintain search input state if no query
+    }
+    setActiveTab(tabId);
+  };
+  const tabsList = [
+    {
+      id: TabKey.YourVaults,
+      label: "Your vaults",
+      disabled: false,
+      handleOnPress: onPress,
+    },
+    {
+      id: TabKey.BrowseLoans,
+      label: "Browse loan tokens",
+      disabled: false,
+      handleOnPress: onPress,
+    },
+  ];
+
+  // Search
+  const [filteredLoans, setFilteredLoans] = useState<LoanToken[]>(loans);
+  const [isVaultReady, setIsVaultReady] = useState(false);
+  const [showSeachInput, setShowSearchInput] = useState(false);
+  const [searchString, setSearchString] = useState("");
+  const handleFilter = useCallback(
+    debounce((searchString: string) => {
+      setFilteredLoans(
+        loans.filter((loan) =>
+          loan.token.displaySymbol
+            .toLowerCase()
+            .includes(searchString.trim().toLowerCase())
+        )
+      );
+    }, 500),
+    [loans, hasFetchedLoansData]
+  );
+
+  useEffect(() => {
+    if (loans.length === 0) {
+      return;
+    }
+    handleFilter(searchString);
+  }, [searchString]);
+
+  useEffect(() => {
+    setFilteredLoans(loans);
+  }, [hasFetchedLoansData]);
+
+  useEffect(() => {
+    if (isFocused) {
+      batch(() => {
+        dispatch(fetchVaults({ address, client }));
+        dispatch(fetchLoanTokens({ client }));
+      });
+    }
+  }, [blockCount, address, isFocused]);
+
+  useEffect(() => {
+    dispatch(fetchLoanSchemes({ client }));
+  }, []);
+
+  useEffect(() => {
+    setIsVaultReady(
+      vaultsList.some((record: any) => record.vaultState !== VaultStatus.Empty)
```

Comment:
```suggestion
      vaultsList.some((vault) => vault.vaultState !== VaultStatus.Empty)
```
For the type, do you agree to remove the `as LoanVault[];` in the `vaultsSelector` because it is misleading forcing a type if it isn't the same with the response

Reply:
yes haha